fscollections/views.py

```diff
@@ -0,0 +1,279 @@
+    user_collections = user_collections.distinct().order_by('modified')
+    try:
+        last_collection = \
+            Collection.objects.filter(user=request.user).order_by('-modified')[0]
+        # If user has a previous bookmark, use the same category by default (or use none if no category used in last
+        # bookmark)
+    except IndexError:
+        last_collection = None
+        
+    if not request.GET.get('ajax'):
+        HttpResponseRedirect(reverse("sound", args=[sound.user.username,sound.id]))
+
+    if request.method == 'POST':
+        form = SelectCollectionOrNewCollectionForm(request.POST, sound_id=sound_id, user_collections=user_collections, user_saving_sound=request.user)
+        if form.is_valid():
+            saved_collection = form.save()
+            CollectionSound.objects.create(user=request.user, collection=saved_collection, sound=sound, status="OK")
+            msg_to_return = f'Sound "{sound.original_filename}" saved under collection {saved_collection.name}'
+            return JsonResponse({'success': True, 'message': msg_to_return})
+    else:
+        form = SelectCollectionOrNewCollectionForm(initial={'collection': last_collection.id if last_collection else SelectCollectionOrNewCollectionForm.BOOKMARK_COLLECTION_CHOICE_VALUE},
+                                sound_id=sound.id,
+                                user_collections=user_collections,
+                                user_saving_sound=request.user)
+    collections_already_containing_sound = user_collections.filter(sounds__id=sound.id).distinct()
+    full_collections = Collection.objects.filter(num_sounds__gte=settings.MAX_SOUNDS_PER_COLLECTION) 
+    tvars = {'user': request.user,
+             'sound': sound,
+             'sound_is_moderated_and_processed_ok': sound.moderated_and_processed_ok,
+             'form': form,
+             'collections_with_sound': collections_already_containing_sound,
+             'full_collections': full_collections
+             }
+
+    return render(request, 'collections/modal_add_sound_to_collection.html', tvars)
+
+def create_collection(request):
+    if not request.GET.get('ajax'):
+      return HttpResponseRedirect(reverse("your-collections"))
+    if request.method == "POST":
+        form = CreateCollectionForm(request.POST, user=request.user)
+        if form.is_valid():
+            Collection.objects.create(user = request.user, 
+                    name = form.cleaned_data['name'], 
+                    description = form.cleaned_data['description'])
+            return JsonResponse({'success': True})
+    else:
+        form = CreateCollectionForm(user=request.user)
+    tvars = {'form': form}
+    return render(request, 'collections/modal_create_collection.html', tvars)
+    
+def delete_collection(request, collection_id):
+    collection = get_object_or_404(Collection, id=collection_id)
+
+    if request.method=="POST" and request.user==collection.user:
+        collection.delete()
```

Comment:
Add a message like "collection successfully deleted"? See how django messages app works, we do it in many other places